src/modules/report/plain.py

```diff
@@ -14,15 +14,21 @@ class PlainReporter(object):
     def get_report(self):
         """generates report tables"""
         output = ""
+        services_lock.acquire()
```

Comment:
Given there are two locks here, I would definitely prefer this code to only hold each lock briefly to get the length. Having a one-to-one relationship between acquire() and release() makes it easier to read the code. It also means you don't ever have to hold both locks at a time.